### PR TITLE
adapt TemporalRowTimeJoinOperator to the new temporal states

### PR DESCRIPTION
**Pull request by NicoK**

### CODE REVIEW

File: flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/TemporalRowTimeJoinOperator.java [hunk 1]
```diff
@@ -82,7 +73,7 @@
 
     private static final long serialVersionUID = 6642514795175288193L;
 
-    private static final String NEXT_LEFT_INDEX_STATE_NAME = "next-index";
+    // TODO: do we have to rename states so Flink fails early when someone tries to restore from old state?
```
Comment (sjwiesman) [inline flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/TemporalRowTimeJoinOperator.java:76]: No, table api doesn't support upgrades yet :)

Comment (NicoK) [reply flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/TemporalRowTimeJoinOperator.java:76]: question is how well it fails if we keep these names - are you confident that I can just remove the comment?

**Comment by NicoK** [comment]
`TemporalRowTimeJoinOperatorTest` is passing now, so it should be ready to go

**Comment by NicoK** [comment]
I'll merge as is - in a final PR, let's have the SQL team decide on the question regarding upgrades - I want to make sure that restoring from incompatible state fails and doesn't silently go into undefined behaviour

**Pull request merged**